routing/index_graph_starter.hpp

```diff
@@ -24,33 +24,28 @@ class IndexGraphStarter final
   {
   public:
     FakeVertex(NumMwmId mwmId, uint32_t featureId, uint32_t segmentIdx, m2::PointD const & point)
-      : m_featureId(featureId), m_segmentIdx(segmentIdx), m_mwmId(mwmId), m_point(point)
+      : m_segment(mwmId, featureId, segmentIdx, true /* forward */), m_point(point)
     {
     }
 
     FakeVertex(Segment const & segment, m2::PointD const & point)
-      : m_featureId(segment.GetFeatureId())
-      , m_segmentIdx(segment.GetSegmentIdx())
-      , m_mwmId(segment.GetMwmId())
-      , m_point(point)
+      : m_segment(segment), m_point(point)
     {
     }
 
-    NumMwmId GetMwmId() const { return m_mwmId; }
-    uint32_t GetFeatureId() const { return m_featureId; }
-    uint32_t GetSegmentIdx() const { return m_segmentIdx; }
+    NumMwmId GetMwmId() const { return m_segment.GetMwmId(); }
+    uint32_t GetFeatureId() const { return m_segment.GetFeatureId(); }
+    uint32_t GetSegmentIdx() const { return m_segment.GetSegmentIdx(); }
+    Segment const & GetSegment() const { return m_segment; }
     m2::PointD const & GetPoint() const { return m_point; }
-
     bool Fits(Segment const & segment) const
     {
-      return segment.GetFeatureId() == m_featureId && segment.GetSegmentIdx() == m_segmentIdx &&
-             segment.GetMwmId() == m_mwmId;
+      return segment.GetFeatureId() == GetFeatureId() &&
```

Comment:
Please, re-order. Compare mwm id first, feature id second, and segment idx last.